From 7c6b97c87004cfb98ef9ae1ebb11906ecaf6be93 Mon Sep 17 00:00:00 2001
From: Tingguo Cheng <quic_tingguoc@quicinc.com>
Date: Fri, 26 Jul 2024 17:38:21 +0800
Subject: [PATCH 20/30] pmdomain: qcom: rpmhpd: sa7255p add power domains

Add power domain description for platofrom sa7255p with new compatible.

Signed-off-by: Shazad Hussain <quic_shazhuss@quicinc.com>
Signed-off-by: Tingguo Cheng <quic_tingguoc@quicinc.com>
---
 drivers/pmdomain/qcom/rpmhpd.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/drivers/pmdomain/qcom/rpmhpd.c b/drivers/pmdomain/qcom/rpmhpd.c
index d2cb4271a1ca..c1f8323bfc8c 100644
--- a/drivers/pmdomain/qcom/rpmhpd.c
+++ b/drivers/pmdomain/qcom/rpmhpd.c
@@ -216,6 +216,29 @@ static struct rpmhpd gmxc = {
 	.res_name = "gmxc.lvl",
 };
 
+/* SA7255P RPMH power domains */
+static struct rpmhpd *sa7255p_rpmhpds[] = {
+	[SA7255P_CX] = &cx,
+	[SA7255P_CX_AO] = &cx_ao,
+	[SA7255P_EBI] = &ebi,
+	[SA7255P_GFX] = &gfx,
+	[SA7255P_LCX] = &lcx,
+	[SA7255P_LMX] = &lmx,
+	[SA7255P_MMCX] = &mmcx,
+	[SA7255P_MMCX_AO] = &mmcx_ao,
+	[SA7255P_MXC] = &mxc,
+	[SA7255P_MXC_AO] = &mxc_ao,
+	[SA7255P_MX] = &mx,
+	[SA7255P_MX_AO] = &mx_ao,
+	[SA7255P_NSP0] = &nsp0,
+	[SA7255P_NSP1] = &nsp1,
+};
+
+static const struct rpmhpd_desc sa7255p_desc = {
+	.rpmhpds = sa7255p_rpmhpds,
+	.num_pds = ARRAY_SIZE(sa7255p_rpmhpds),
+};
+
 /* SA8540P RPMH powerdomains */
 static struct rpmhpd *sa8540p_rpmhpds[] = {
 	[SC8280XP_CX] = &cx,
@@ -625,6 +648,7 @@ static const struct rpmhpd_desc x1e80100_desc = {
 
 static const struct of_device_id rpmhpd_match_table[] = {
 	{ .compatible = "qcom,qdu1000-rpmhpd", .data = &qdu1000_desc },
+	{ .compatible = "qcom,sa7255p-rpmhpd", .data = &sa7255p_desc },
 	{ .compatible = "qcom,sa8155p-rpmhpd", .data = &sa8155p_desc },
 	{ .compatible = "qcom,sa8540p-rpmhpd", .data = &sa8540p_desc },
 	{ .compatible = "qcom,sa8775p-rpmhpd", .data = &sa8775p_desc },
-- 
2.34.1

